### PR TITLE
Add synthetic source support for binary fields

### PR DESCRIPTION
**Pull request by lkts**
This PR adds synthetic source support for `binary` fields. Synthetic source is only supported when doc values are enabled.

Contributes to #106460.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/BinaryDocValuesSyntheticFieldLoader.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.LeafReader;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public abstract class BinaryDocValuesSyntheticFieldLoader implements SourceLoader.SyntheticFieldLoader {
```
Comment (lkts) [inline server/src/main/java/org/elasticsearch/index/mapper/BinaryDocValuesSyntheticFieldLoader.java:20]: This is introduced in #107081 which is not yet merged.

**Comment by lkts** [comment]
@elasticmachine update branch

**Comment by lkts** [comment]
@elasticmachine update branch

**Comment by lkts** [comment]
@elasticmachine update branch

**Review by martijnvg — APPROVED** [review]
LGTM

#107554 part is missing, but is required the fully support synthetic source for the binary field type. This will be added in a follow up.

**Pull request merged**